R/add_item.R

```diff
@@ -81,48 +81,35 @@ add_item <- function(
   # fetch the host from the token
   host <- token[["arcgis_host"]]
 
-  # if async = TRUE stop
-  # type must be feature service right now
-  # TODO make this cli_abort()
-  stopifnot(
-    "`async` must be `FALSE`" = !async,
-    "`type` must be `\"Feature Service\"`" = identical(type, "Feature Service")
+  check_add_item_args(
+    description = description,
+    snippet = snippet,
+    async = async,
+    type = type
   )
 
   # if CRS is missing require user input if interactive
   if (interactive() && is.na(sf::st_crs(x)) && inherits(x, "sf")) {
-    choice <- utils::menu(
-      c("Yes", "No"),
-      title = "CRS is missing from `x`. Continue?"
-    )
+    cli::cli_bullets(c("!" = "{.arg x} has no CRS"))
 
-    if (choice == 2L) {
-      # TODO cli_abort
-      stop("Aborting. CRS is missing.")
-    } else {
-      # TODO cli_warn
-      warning("Set the CRS to prevent this interruption.\n  - use `sf::st_set_crs()`")
+    choice <- readline(paste0("?\u00a0Continue?\u00a0(Y/n)\u00a0"))
+
+    if (choice %in% c("n", "N", "no", "0")) {
+      cli::cli_abort("Aborting.")
     }
+
```

Comment:
This lets users type anything and does not re-prompt for a bad value. Plus I can foresee the use of unicode characters as causing issues on some windows machines. I'd like to revert this.